revolsys-core/src/main/java/com/revolsys/record/schema/TableRecordStoreInsertUpdateBuilder.java

```diff
@@ -30,30 +30,24 @@ public Record executeDo() {
         final Record newRecord = newRecord();
         if (newRecord == null) {
           return null;
-        } else {
-          insertRecord(newRecord);
-          try {
-            return this.recordStore.insertRecord(this.connection, newRecord);
-          } catch (final Exception e) {
-            throw Exceptions.wrap("Unable to insert record:\n" + newRecord, e);
-          }
         }
-      } else {
-        return null;
-      }
-    } else {
-      if (isUpdate()) {
+        insertRecord(newRecord);
         try {
-          updateRecord(changeTrackRecord);
-          this.recordStore.updateRecordDo(this.connection, changeTrackRecord);
-          return changeTrackRecord.newRecord();
-        } catch (final Exception e) {
-          throw Exceptions.wrap("Unable to update record:\n" + changeTrackRecord, e);
+          return this.recordStore.insertRecord(this.connection, newRecord);
+        } catch (final RuntimeException e) {
+          throw Exceptions.wrap("Unable to insert record:\n" + newRecord, e);
         }
-      } else {
-        return null;
+      }
+    } else if (isUpdate()) {
+      try {
+        updateRecord(changeTrackRecord);
```

Comment:
should we rename that updateRecord method to prepareUpdateRecord. It confused me reading my code again why there was to function like calls?